docs/ref/checks.txt

```diff
@@ -664,4 +664,4 @@ The following checks are performed on your URL configuration:
   including a ``:``. Remove the colon, to avoid ambiguous namespace
   references.
 * **urls.E004**: Your URL pattern ``<pattern>`` is invalid. Ensure that
-  ``urlpatterns`` is a list of :func:`~django.conf.urls.url()` instances.
+  ``urlpatterns`` is a list of :func:`~django.urls.path()` instances.
```

Comment:
Right now the consensus has been "make `path(...)` the default documented usage, so I've started by transfering everything across to that.

I've got "Add docs for `path_regex()`" on the TODO list. Need to figure out how & where best to include that.